src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.Reference.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.AddImport
+{
+    internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSyntax>
+    {
+        private abstract class Reference : IComparable<Reference>, IEquatable<Reference>
+        {
+            protected readonly AbstractAddImportCodeFixProvider<TSimpleNameSyntax> provider;
+            public readonly SearchResult SearchResult;
+
+            protected Reference(AbstractAddImportCodeFixProvider<TSimpleNameSyntax> provider, SearchResult searchResult)
+            {
+                this.provider = provider;
+                this.SearchResult = searchResult;
+            }
+
+            public virtual int CompareTo(Reference other)
+            {
```

Comment:
"null" would indicate a bug somewhere in our code.  As such, i'd prefer to crash and get a dump to work with rather than silently moving on when this happens.  Note that this is not a public type, but something we control completely.